dbt_platform_helper/providers/ecs.py

```diff
@@ -53,6 +71,45 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def wait_until(
+    exceptions_to_catch: tuple = (PlatformException,),
+    max_attempts: int = RETRY_MAX_ATTEMPTS,
+    delay: int = SECONDS_BEFORE_RETRY,
+    raise_custom_exception: bool = True,
+    custom_exception=RetryException,
+    message_on_false="Condition not met",
+):
+    """Wrap a function which returns a boolean."""
+
+    def decorator(func: Callable[..., bool]):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            last_exception = None
+            for attempt in range(max_attempts):
+                try:
+                    result = func(*args, **kwargs)
+                except exceptions_to_catch as e:
+                    result = False
+                    last_exception = e
+
+                if result:
+                    return result
+
+                if attempt < max_attempts - 1:
+                    time.sleep(delay)
+
+            # TODO is this the way we want it?
```

Comment:
I think this works. It gives the flexibility to retry on exception or on a false result.